lib/activerecord_spanner_adapter/transaction.rb

```diff
@@ -65,11 +80,17 @@ def rollback
       # Allow rollback after abort and/or a failed commit.
       raise "This transaction is not active" unless active? || @state == :FAILED || @state == :ABORTED
       if active?
-        @connection.session.rollback @grpc_transaction.transaction_id unless @isolation == :read_only
+        shoot_and_forget_rollback
```

Comment:
So looks like rollback here is only best effort. Can you comment a bit here for the why?

Reply:
Added comments on why a shoot-and-forget rollback is done.